src/object/object_primitive.c

```diff
@@ -17456,12 +17459,35 @@ mr_data_writeval_json (OR_BUF * buf, DB_VALUE * value)
   int rc = NO_ERROR;
   DB_VALUE json_body, schema_raw;
 
+  db_make_null (&json_body);
+  db_make_null (&schema_raw);
+
   if (value->data.json.json_body == NULL || DB_IS_NULL (value))
     {
       assert (false);
       return ER_FAILED;
     }
 
+  if (buf->error_abort)
+    {
+      int estimated_length = mr_data_lengthval_json (value, true);
+
+      /* in case of string compression, buffer will hold
+       * 0xFF int int as first bytes, so we need to take them
+       * in consideration
+       */
+      estimated_length += sizeof (char) + sizeof (int) * 2;
```

Comment:
Doesn't mr_data_lengthval_json  already count the length header?

Reply:
you're right